### PR TITLE
Add io.github.insilmaril.vym.appdata.xml

### PR DESCRIPTION
**Pull request by bztd**
Needed for flathub.

I put in the description what I thought was correct, it would be good to review it.
The images do not have a good resolution, they should be changed.

### CODE REVIEW

**Comment by insilmaril** [comment]
Thanks for the flathub config!

I wonder if we also could use OBS (Open Build Service) to automatically create flatpak packages - I uses it for various SUSE, Redhat, Ubuntu, ...  packaging already and it flatpak should be supported by now. Ideally with one "obs build" all platforms would be created automatically. An example package could be "kraft", didn't have time to look into the details, though. What do you think?

**Comment by bztd** [comment]
You don't use many dependencies beyond Qt, maintaining the package is very easy, I can help, but it's always good for the developer to have control. As for your question, I don't know, I would think not. The only automations I know of are using a bot, and github actions.

Apart from that, I think vym should not use dbus in general, this feature can go beyond the sandbox. You can be specific and point out which dbus is used, I'm looking into that option. If dbus is used it's very likely that the application will not be allowed. Permissions can be edited by users and they are the ones in control.

(this is a translation)

**Comment by bztd** [comment]
Flathub runs its own packaging system: https://buildbot.flathub.org , I don't think they allow packaging from any other source. The only exception I've seen is firefox.

**Comment by insilmaril** [comment]
ideally config could also be used to build both on flathub and OBS, but I have no experience with flatpak/hub.

For DBUS: It's added only when found during build (see CMakeList.txt), e.g. on Mac and Windows it's not used. Not sure if it would be found on flathub?

I can't really comment on the flathub file, as I haven't tried that yet - anything specific I could help with?

And would you kind of commit to maintain it? ;-)

**Comment by bztd** [comment]
dbus is accessible as long as the appropriate permissions are granted. I'll see if support can be maintained or removed.

"anything specific I could help with?" 
* Mainly examine the package description, change it if you wish.
* It is also necessary to add the application icon in 128x128 minimum and 512x512 maximum or .svg format in the default path for future versions.
* I have not been able to find the source for version 2.9.27 available on sourceforge.


"And would you kind of commit to maintain it? "
Yes, I can help with that.

The link to the petition to add vym to flathub: https://github.com/flathub/flathub/pull/5534. The package does not work because it is missing the mentioned permission.

File: config/io.github.insilmaril.vym.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    <id>io.github.insilmaril.vym</id>
+    <metadata_license>CC0-1.0</metadata_license>
+    <project_license>LGPL-2.1-or-later</project_license>
+    <name>vym</name>
+    <summary>Tool for generating and manipulating mind maps</summary>
+    <description>
+	<p>
+	    VYM (View Your Mind) allows you to easily and intuitively
+            organize your ideas in a structured way. You can create 
+            metal maps and export them to multiple formats.
```
Comment (insilmaril) [inline config/io.github.insilmaril.vym.appdata.xml]: "mental" maps

Maybe also worth mentioning: "vym is also powerful personal task manager"

File: config/io.github.insilmaril.vym.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	</p>
+    </description>
+    <screenshots>
+        <screenshot type="default">
+          <image>https://a.fsdn.com/con/app/proj/vym/screenshots/328319.jpg</image>
```
Comment (insilmaril) [inline config/io.github.insilmaril.vym.appdata.xml:17]: Meanwhile we could have nicer screenshots, but I guess these still can be updated later ;-)

**Review by insilmaril — CHANGES_REQUESTED** [review]
Can you please fix the "metal" typo?

File: config/io.github.insilmaril.vym.appdata.xml (hunk 2 above)
Comment (bztd) [reply config/io.github.insilmaril.vym.appdata.xml:17]: The package has already been incorporated into flathub, the changes will have to be made in the next update.

**Comment by insilmaril** [comment]
Thanks!

**Pull request merged**